cmd/transcriber/transcribe/output.go

```diff
@@ -4,15 +4,33 @@ import (
 	"fmt"
 	"io"
 	"math"
+	"regexp"
 	"sort"
 	"strings"
 )
 
+var (
+	segmentSanitizationSpacesRE = regexp.MustCompile(`\s+`)
+	// We allow spaces, dots, digits and letters in both ASCII and foreign alphabets.
+	segmentSanitizationSpecialRE = regexp.MustCompile(`[^\s\d\pL\pN.]`)
+)
+
 type namedSegment struct {
 	Segment
 	Speaker string
 }
 
+func (ns *namedSegment) sanitize() {
+	// Remove unwanted special characters
+	ns.Speaker = segmentSanitizationSpecialRE.ReplaceAllString(ns.Speaker, "")
```

Comment:
I feel like we should do something like this for `ns.Text` as well, just in case. At least filter out angle brackets and parens. The model shouldn't produce any of those, but you never know... unless there are already some safeguards in place that I'm not aware of?

Reply:
I was mostly worried we could remove non speech tokens and symbols that the model can output at times (e.g. silence, music, laugh) so I probably need to do some more research to avoid being too strict.

Reply:
Looks like WebVTT supports HTML character references, so for that one, `html.EscapeString` is the right approach.

Reply:
@jupenur So this is the list of the non speech tokens:

```
static const std::vector<std::string> non_speech_tokens = {
    "\"", "#", "(", ")", "*", "+", "/", ":", ";", "<", "=", ">", "@", "[", "\\", "]", "^",
    "_", "`", "{", "|", "}", "~", "「", "」", "『", "』", "<<", ">>", "<<<", ">>>", "--",
    "---", "-(", "-[", "('", "(\"", "((", "))", "(((", ")))", "[[", "]]", "{{", "}}", "♪♪",
    "♪♪♪","♩", "♪", "♫", "♬", "♭", "♮", "♯"
};
```

https://github.com/ggerganov/whisper.cpp/blob/fa8dbdc8886150254748854387ea9240385800e2/whisper.cpp#L3455C1-L3460C3

There's also a parameter to disable them entirely if we want. I am not too opinionated here, let me know.

Reply:
With WebVTT we should be ok if we just allow everything but escape it properly. The other format I'm not sure about. Does the client use it for something, or is it just a transcript for direct human consumption?

Reply:
Okay. Yes, the text file would be purely for human preview as a normal file attachment.

Reply:
Ok then I don't see any big issues there either. We can just leave that one as it is and fix WebVTT by adding the escaping.

Reply:
Sounds good, thanks :+1: